General `const` correctness

aromanielloNTIA opened the pull request:
Prompted by #9, this pull request retrofits the ITM codebase with [`const` correctness](https://isocpp.org/wiki/faq/const-correctness) wherever applicable. I used a [static code analysis tool](https://github.com/danmar/cppcheck) to evaluate the ITM and ITMDrvr source code, and then modified the code to use `const` as much as possible. I tested the changes by compiling and using ITMDrvr to run the included example cases, to verify outputs have not changed.

Note for @wkozmaNTIA: I did not modify the C#/.NET wrapper to account for these changes, nor did I modify any version numbers. I'll leave these to you.

aromanielloNTIA committed:
const correctness

blainNTIA commented on include/itm.h:
```diff
@@ -69,37 +69,37 @@ DLLEXPORT int ITM_AREA_CR_Ex(double h_tx__meter, double h_rx__meter, int tx_site
+DLLEXPORT void LinearLeastSquaresFit(const double pfl[], double d_start, double d_end, double *fit_y1, double *fit_y2);
+DLLEXPORT double LineOfSightLoss(double d__meter, const double h_e__meter[2], complex<double> Z_g, double delta_h__meter,
     double M_d, double A_d0, double d_sML__meter, double f__mhz);
 DLLEXPORT int LongleyRice(double theta_hzn[2], double f__mhz, complex<double> Z_g, double d_hzn__meter[2], double h_e__meter[2], 
```
```suggestion
DLLEXPORT int LongleyRice(const double theta_hzn[2], double f__mhz, complex<double> Z_g, const double d_hzn__meter[2], const double h_e__meter[2], 
```

blainNTIA reviewed (approved):
Great work! There might be a couple of other places where adding "const" would be good. Please review my comments.

aromanielloNTIA committed:
more const correctness

aromanielloNTIA committed:
Merge branch 'master' into const-correctness

aromanielloNTIA committed:
even more const correctness

aromanielloNTIA committed:
main function const correctness

aromanielloNTIA commented:
@blainNTIA looking back at this, with much more C++ experience compared to when I made the initial changes, I found many more instances where const correctness could be implemented. The static code analysis tool I initially used did not identify these cases, but I went through the functions to determine where `const` was applicable. The changes I made included your suggestions and similar changes in many other places. Please take a look at the new changes and provide an updated review when you get a chance.

aromanielloNTIA commented:
Also, I still have not made edits to the .NET wrapper, or tested the .NET wrapper for the C++ changes. I don't think any changes are needed in .NET based on the addition of `const` in the C++ code, but we should test this before merging this PR.

aromanielloNTIA committed:
Make variability parameters TLS/CR const

time, location, and situation parameters were previously non-const since the percentage inputs were overwritten with ratios inside the main P2P and Area mode Ex entrypoint functions. Modified the use of these variables in Variability.cpp so they are scaled to ratios when used, and the inputs are no longer modified.

aromanielloNTIA committed:
Make internal np variable const

aromanielloNTIA committed:
Make internal p10 variable const

p10 is not modified

aromanielloNTIA committed:
Add const to ComputeDeltaH variables

aromanielloNTIA committed:
Add const to DiffractionLoss variables

aromanielloNTIA committed:
Add const to FindHorizons variables

aromanielloNTIA committed:
Add const to H0Function

aromanielloNTIA committed:
Add const to InitializeArea

Also use constexpr for a variable which is always known at compile time

aromanielloNTIA committed:
Add const to InitializePointToPoint

And constexpr for gamma_a which is always known at compile time

aromanielloNTIA committed:
Add const to ICCDF

Also convert many constants to constexpr to ensure compile-time evaluation

aromanielloNTIA committed:
Add const to d__meter in itm_area

aromanielloNTIA committed:
Add internal const to KnifeEdgeDiffraction

aromanielloNTIA committed:
Add internal const to LinearLeastSquaresFit

aromanielloNTIA committed:
Add internal const to LineOfSightLoss

aromanielloNTIA committed:
Add internal const to LongleyRice

aromanielloNTIA committed:
Add internal const to QuickPfl

aromanielloNTIA committed:
Add internal const to SmoothEarthDiffraction

aromanielloNTIA committed:
Add internal const to TroposcatterLoss

Also use constexpr for Z_0, Z_1 values since they should be evaluated at compile time

aromanielloNTIA committed:
Add internal const to Variability

Also change many constant arrays to constexpr since they are not runtime dependent.

aromanielloNTIA committed:
Make mdvar constant

Use an internal copy in Variability.cpp to modify based on the +20 and +10 modifiers for different modes

aromanielloNTIA committed:
Make climate enum input const

Use an internal copy which is modified in Variability.cpp

aromanielloNTIA committed:
Make mdvar const in header

Finishing off the change in 54b60a7c6f45d7cf54244b2b7780d03113e5db25
